File: backend/app/schemas/onboarding.py
"""
Pydantic schemas for onboarding endpoints.
"""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class OnboardingStepUpdate(BaseModel):
    """Schema for updating a specific onboarding step."""

    step: int = Field(..., ge=1, le=6, description="Step number (1-6)")
    completed: bool = Field(default=True, description="Whether the step is completed")


class OnboardingSkip(BaseModel):
    """Schema for skipping onboarding."""

    notes: Optional[str] = Field(
        None, description="Optional notes about why onboarding was skipped"
    )


class OnboardingResponse(BaseModel):
    """Schema for onboarding progress response."""

    id: UUID
    user_id: UUID
    is_completed: bool
    is_skipped: bool
    step_profile_completed: bool
    step_content_preferences_completed: bool
    step_settings_completed: bool
    step_my_posts_completed: bool
    step_content_ideas_completed: bool
    step_posting_schedule_completed: bool
    current_step: int
    progress_percentage: float
    notes: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    skipped_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class OnboardingCreate(BaseModel):
    """Schema for creating onboarding progress."""

    user_id: str
    current_step: int = Field(default=1, ge=1, le=6)
    notes: Optional[str] = None


class OnboardingUpdate(BaseModel):
    """Schema for updating onboarding progress."""

    current_step: Optional[int] = Field(None, ge=1, le=6)
    notes: Optional[str] = None
    step_profile_completed: Optional[bool] = None
    step_content_preferences_completed: Optional[bool] = None
    step_settings_completed: Optional[bool] = None
    step_my_posts_completed: Optional[bool] = None
    step_content_ideas_completed: Optional[bool] = None
    step_posting_schedule_completed: Optional[bool] = None

File: frontend/src/types/onboarding.ts
/**
 * Types for onboarding functionality
 */

export interface OnboardingProgress {
  id: string;
  user_id: string;
  is_completed: boolean;
  is_skipped: boolean;
  step_profile_completed: boolean;
  step_content_preferences_completed: boolean;
  step_settings_completed: boolean;
  step_my_posts_completed: boolean;
  step_content_ideas_completed: boolean;
  step_posting_schedule_completed: boolean;
  current_step: number;
  progress_percentage: number;
  notes?: string;
  created_at: string;
  updated_at?: string;
  completed_at?: string;
  skipped_at?: string;
}

export interface OnboardingStepUpdate {
  step: number;
  completed?: boolean;
}

export interface OnboardingSkip {
  notes?: string;
}

export interface OnboardingUpdate {
  current_step?: number;
  notes?: string;
  step_profile_completed?: boolean;
  step_content_preferences_completed?: boolean;
  step_settings_completed?: boolean;
  step_my_posts_completed?: boolean;
  step_content_ideas_completed?: boolean;
  step_posting_schedule_completed?: boolean;
}

export interface OnboardingStep {
  id: number;
  title: string;
  icon: string;
  isCompleted: boolean;
  isActive: boolean;
  highlights?: string[];
}

export const ONBOARDING_STEPS: Omit<
  OnboardingStep,
  "isCompleted" | "isActive"
>[] = [
  {
    id: 1,
    title: "Set Up Your Profile",
    icon: "👤",
    highlights: [
      "Add your LinkedIn handle (required)",
      "Add your Substack handle (optional)",
      "Click Analyze to analyze your bio, writing style, and preferences (takes about 5 minutes)",
    ],
  },
  {
    id: 2,
    title: "Content Preferences",
    icon: "📝",
    highlights: [
      "Enter topics you like to write about",
      "Add news websites you follow",
    ],
  },
  {
    id: 3,
    title: "Settings",
    icon: "⚙️",
    highlights: ["Set when you want to see daily suggested drafts"],
  },
  {
    id: 4,
    title: "My Posts",
    icon: "📄",
    highlights: [
      "Use 'New Post' button to create posts manually",
      "Use 'Brainstorm' button to brainstorm drafts with AI",
      "The AI will improve as your write more posts and provide feedback",
    ],
  },
  {
    id: 5,
    title: "Content Ideas",
    icon: "💡",
    highlights: [
      "Use 'Add idea' button to drop down a few notes or drop in a URL to an article",
      "Use 'Brainstorm' button to brainstorm drafts with AI using an idea",
    ],
  },
];
